core/amber/src/main/scala/edu/uci/ics/amber/engine/common/promise/PromisePayload.scala

```diff
@@ -0,0 +1,12 @@
+package edu.uci.ics.amber.engine.common.promise
+
+import edu.uci.ics.amber.engine.common.ambermessage.neo.ControlPayload
+
+/** For now, promise is only one type of control
+  * message. I reserved some space for other control
+  * message. If no need, ControlPayload can be
+  * replaced by this class.
+  */
+trait PromisePayload extends ControlPayload {
```

Comment:
we have to think of another name here. Every promise has its own payload -  the promised payload - the one that you set by calling setValue

Reply:
Can we just use ControlPayload?

Reply:
I think that will be better

Reply:
I think ControlPayload is the accurate name. A control payload has (ctx,message)

Reply:
Is status report a kind of control message? If so, I think we need to keep both ControlPayload and PromisePayload because status reports do not require any reply.